File: Python/1221.py
# 1221. Split a String in Balanced Strings
# https://leetcode.com/problems/split-a-string-in-balanced-strings/

class Solution:
  def balancedStringSplit(self, s: str) -> int:
    out = 0
    _sum = 0

    for _, c in enumerate(s):
      _sum += 1 if (c == "R") else -1

      if (_sum == 0):
        out += 1

    return out

File: Rust/1221.rs
/**
 * 1221. Split a String in Balanced Strings
 * https://leetcode.com/problems/split-a-string-in-balanced-strings/
 */

impl Solution {
  pub fn balanced_string_split(s: String) -> i32 {
    let mut out: i32 = 0;
    let mut sum: i32 = 0;

    for c in s.chars() {
      sum += if c == 'R' { 1 } else { -1 };

      if sum == 0 {
        out += 1;
      }
    }

    return out;
  }
}
